packages/mgt-components/src/graph/graph.groups.ts

```diff
@@ -142,16 +147,39 @@ export async function findGroups(
       filterGroups.push('(mailEnabled eq true and securityEnabled eq false)');
     }
 
-    filterQuery += (query !== '' ? ' and ' : '') + filterGroups.join(' or ');
-  }
+    if (query !== '') {
+      filterQuery = filterQuery + ' and ';
+    }
 
-  const result = await graph.api('groups').filter(filterQuery).top(top).middlewareOptions(prepScopes(scopes)).get();
+    for (let filter of filterGroups) {
+      batch.get(filter, `/groups?$filter=${filterQuery + filter}`, ['Group.Read.All']);
+    }
 
-  if (getIsGroupsCacheEnabled() && result) {
-    cache.putValue(key, { groups: result.value.map(x => JSON.stringify(x)), top: top });
+    responses = await batch.executeAll();
```

Comment:
This should probably be wrapped in a try catch and including a fall back promise calls.
```
try {
    // do batch
    // return batchedResult;
} catch (_) {
    try {
        return Promise.all(// make individual calls instead of batch in case batch fails)
    } catch (_) {
        return [];
    }
}
```
[See example here](https://github.com/microsoftgraph/microsoft-graph-toolkit/blob/87be56d08cfd3688c08a8c13841086e060f4899e/packages/mgt-components/src/graph/graph.files.ts#L953)

Reply:
Sorry for the delay I was trying to figure this out.. See here:
53093b4